tests/ci/check_so_interface.py

```diff
@@ -276,68 +303,94 @@
 iv_files - generated by run_iv"
 """
 
-def compare_args(node, intdict, name):
-    #check function
-    if isinstance(node, ast.FunctionDef):
-        spec_args = intdict['args']
-        spec_defaults = intdict["defaults"]
-
-        # Extract function arguments from the AST node
-        found_args = [arg.arg for arg in node.args.args]
-        found_defaults = [ast.unparse(d) if d else None for d in node.args.defaults]
-
-        has_varargs = any(isinstance(arg, ast.arg) and arg.arg == 'args' for arg in node.args.args) #check for varargs
-
-        has_kwargs = any(isinstance(arg, ast.arg) and arg.arg == 'kwargs' for arg in node.args.args) #check for kwargs
-
-        if has_varargs == True:
-            raise NotImplementedError("This code doesn't currently have the functionality to support varargs, " + name + " has vararg")
-        if has_kwargs == True:
-            raise NotImplementedError("This code doesn't currently have the functionality to support kwargs, " + name + " has kwarg")
+def compare_args(node, intdict):
+    """
+    Uses intdict to compare the expected 'args', 'defaults', and 'kwargs' 
+    values for certain frozen function specified by node.name to those 
+    values extracted from node, printing feedback on detected differences 
+    and returning false or returning true if no differences detected.
 
-        # Compare arguments to specified arguments in interface dict
-        if found_args != spec_args:
-            print(f"Mismatch in arguments for function {name}")
-            print(f"Found: {found_args}")
-            print(f"Expected: {spec_args}")
-            return False
-        
-        # Compare default values to specified defaults in interface dict
-        if found_defaults != spec_defaults:
-            print(f"Mismatch in default values for function {name}")
-            print(f"Found: {found_defaults}")
-            print(f"Expected: {spec_defaults}")
-            return False
-    #Check class
-    #elif isinstance(node, ast.ClassDef):
+    Args
+    ---
+    node : an AST tree node
+        Node from the AST tree that pairs with a 
+        given function from the interface dictionary.
+    intdict : A dictionary
+        Dictionary that contains values for 'args',
+        'defaults' and 'kwargs' for a given function
 
+    Returns
+    -------
+    Boolean, or raises an error
```

Comment:
should update the docstring if you are returning an error message. (or if the expected behaviour becomes raising an exception on failure)